sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -0,0 +1,505 @@
+    az_iot_hub_client const* client,
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter);
+
+/**
+ * @brief Twin response type.
+ *
+ */
+typedef enum
+{
+  AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_GET = 1,
+  AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES = 2,
+  AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES = 3,
+} az_iot_hub_client_twin_response_type;
+
+/**
+ * @brief Twin response.
+ *
+ */
+typedef struct az_iot_hub_client_twin_response
+{
+  az_iot_hub_client_twin_response_type response_type; /** @brief Twin response type. */
+  az_iot_hub_client_status status; /** @brief The operation status. */
+  az_span request_id; /** @brief Request ID matches the ID specified when issuing a Get or Patch
+                         command. */
+  az_span version; /** @brief The Twin object version.
+                    * @note This is only returned when
+                    * response_type==AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES or
+                    * response_type==AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES */
+} az_iot_hub_client_twin_response;
+
+/**
+ * @brief Attempts to parse a received message's topic.
+ *
+ * @param client
+ * @param received_topic An #az_span containing the received topic.
+ * @param out_request If the message is twin-operation related, this will contain the
+ *                    #az_iot_hub_client_twin_response.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_twin_received_topic_parse(
+    az_iot_hub_client const* client,
+    az_span received_topic,
+    az_iot_hub_client_twin_response* out_twin_response);
+
+/**
+ * @brief Gets the MQTT topic that must be used to submit a Twin GET request.
+ * @note The payload of the MQTT publish message should be empty.
+ *
+ * @param client
+ * @param request_id The request id.
+ * @param mqtt_topic An empty #az_span with sufficient capacity to hold the MQTT topic.
+ * @param out_mqtt_topic The output #az_span containing the MQTT topic.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_twin_get_publish_topic_get(
+    az_iot_hub_client const* client,
+    az_span request_id,
+    az_span mqtt_topic,
+    az_span* out_mqtt_topic);
+
+/**
+ * @brief Gets the MQTT topic that must be used to submit a Twin GET request.
+ * @note The payload of the MQTT publish message should contain a JSON document formatted according
+ *       to the Twin specification.
+ *
+ * @param client
+ * @param request_id The request id.
+ * @param if_match_version Can be either "*" to overwrite or the twin version to limit the reported
+ *                         properties patch to that version only.
+ * @param mqtt_topic An empty #az_span with sufficient capacity to hold the MQTT topic.
+ * @param out_mqtt_topic The output #az_span containing the MQTT topic.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_twin_patch_publish_topic_get(

```

Comment:
>az_iot_hub_client_twin_patch_publish_topic_get [](start = 23, length = 46)

Again if I'm misremembering set me straight, but don't we need a topic for the subscribe for deltas disttinct from what we have?

Reply:
There are 5 APIs in total here:
2x SUB topics - one for responses (from both GET and reported PATCH) and one for deltas (desired PATCH initiated from the server). (`az_iot_hub_client_twin_response_subscribe_topic_filter_get`, `az_iot_hub_client_twin_patch_subscribe_topic_filter_get` respectively)
1x for processing incoming messages `az_iot_hub_client_twin_received_topic_parse`
2x PUB topics - one for GET and one for PATCH. (`az_iot_hub_client_twin_get_publish_topic_get` and `az_iot_hub_client_twin_patch_publish_topic_get` respectively - again in need for some better names).